nixos/modules/services/web-apps/porn-vault.nix

```
@@ -0,0 +1,266 @@
+              };
+              matchCreatedActors = true;
+              matchCreatedStudios = true;
+              matchCreatedLabels = true;
+            };
+            persistence = {
+              backup = {
+                enable = true;
+                maxAmount = 10;
+              };
+              libraryPath = "/media/porn-vault/lib";
+            };
+            plugins = {
+              allowActorThumbnailOverwrite = false;
+              allowMovieThumbnailOverwrite = false;
+              allowSceneThumbnailOverwrite = false;
+              allowStudioThumbnailOverwrite = false;
+              createMissingActors = false;
+              createMissingLabels = false;
+              createMissingMovies = false;
+              createMissingStudios = false;
+              events = {
+                actorCreated = [ ];
+                actorCustom = [ ];
+                sceneCreated = [ ];
+                sceneCustom = [ ];
+                movieCustom = [ ];
+                studioCreated = [ ];
+                studioCustom = [ ];
+              };
+              register = { };
+              markerDeduplicationThreshold = 5;
+            };
+            processing = {
+              generatePreviews = true;
+              readImagesOnImport = false;
+              generateImageThumbnails = true;
+            };
+            server = {
+              https = {
+                certificate = "";
+                enable = false;
+                key = "";
+              };
+            };
+            transcode = {
+              hwaDriver = null;
+              vaapiDevice = "/dev/dri/renderD128";
+              h264 = {
+                preset = "veryfast";
+                crf = 23;
+              };
+              webm = {
+                deadline = "realtime";
+                cpuUsed = 3;
+                crf = 31;
+              };
+            };
+          };
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services.porn-vault = {
+      description = "Porn-Vault server";
+      environment = {
+        PV_CONFIG_FOLDER = "/etc/porn-vault";
+        NODE_ENV = "production";
+        DATABASE_NAME = "production";
+        PORT = toString cfg.port;
+      };
+      serviceConfig = {
+        ExecStart = getExe cfg.package;
+        # Hardening options
+        CapabilityBoundingSet = [ "CAP_SYS_NICE" ];
+        AmbientCapabilities = [ "CAP_SYS_NICE" ];
+        LockPersonality = true;
+        NoNewPrivileges = true;
+        PrivateTmp = true;
+        ProtectControlGroups = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectSystem = true;
+        RestrictNamespaces = true;
+        RestrictSUIDSGID = true;
+        Restart = "on-failure";
+        RestartSec = 0;
+      };
+      wantedBy = mkIf cfg.autoStart [ "multi-user.target" ];
```

```suggestion
      wantedBy = mkIf cfg.autoStart [ "multi-user.target" ];
      wants  = [ "network.target" ];
```
Assuming it needs the network

---

Does it also need network.target if it can run entirely offline on localhost:3000?